modules/tinymce/src/plugins/advlist/main/ts/ui/Buttons.ts

```diff
@@ -26,13 +25,11 @@ const normalizeStyleValue = (styleValue: string | undefined): string =>
   Type.isNullable(styleValue) || styleValue === 'default' ? '' : styleValue;
 
 const makeSetupHandler = (editor: Editor, nodeName: ListType) => (api: Toolbar.ToolbarSplitButtonInstanceApi | Toolbar.ToolbarToggleButtonInstanceApi) => {
-  const nodeChangeHandler = (e: EditorEvent<NodeChangeEvent>) => {
+  const nodeChangeHandler = (e: NodeChangeEvent) => {
     api.setActive(ListUtils.inList(editor, e.parents, nodeName));
     api.setEnabled(!ListUtils.isWithinNonEditableList(editor, e.element));
   };
```

Comment:
Maybe just instead of simulating events you just break up the state change into a new function. Also it needs to use `getStart(true)` to do the same thing as nodeChange.

```suggestion
  const updateButtonState = (element: Element, parents: Node[]) => {
    api.setActive(ListUtils.inList(editor, parents, nodeName));
    api.setEnabled(!ListUtils.isWithinNonEditableList(editor, element));
  };

  const nodeChangeHandler = (e: NodeChangeEvent) => updateButtonState(e.element, e.parent);
  const element = editor.selection.getStart(true);
  updateButtonState(element, editor.dom.getParents(element));
```

Reply:
changed